media/js/base/mozilla-monitor-button.js

```diff
@@ -20,17 +20,27 @@ Mozilla.MonitorButton = (function() {
     }
 
     var buttonURL = monitorButton.getAttribute('href');
+    // strip url to everything after `?`
+    var buttonURLParams = buttonURL.match(/\?(.*)/)[1];
+
     var destURL = monitorButton.getAttribute('data-action') + 'metrics-flow';
 
+    // collect values from monitor button
+    var params = window._SearchParams.queryStringToObject(buttonURLParams);
+
+    // add required params to the token fetch request
+    destURL += '&' + params;
```

Comment:
This is passing an object to the metrics flow, so the request comes out like `/metrics-flow&[object%20Object]`, which results in a 404.

I'd also suggest we only pass the specific params in the issue description, vs everything a URL query string may contain?